railties/test/generators/app_generator_test.rb

```diff
@@ -259,6 +259,20 @@ def test_config_jdbc_database_when_no_option_given
     end
   end
 
+  def test_generator_without_skips
+    run_generator
+    assert_file "config/application.rb", /\s+require\s+["']rails\/all["']/
+    assert_file "config/environments/development.rb" do |content|
+      assert_match(/config\.action_mailer\.raise_delivery_errors = false/, content)
+    end
+    assert_file "config/environments/test.rb" do |content|
+      assert_match(/config\.action_mailer\.delivery_method = :test/, content)
+    end
+    assert_file "config/environments/production.rb" do |content|
+      assert_match(/# config\.action_mailer\.raise_delivery_errors = false/, content)
+    end
+  end
```

Comment:
@carlosantoniodasilva 

Before this PR, generating an app without `--skip-active-record`, `--skip-sprockets` or `--skip-test-unit` would add `require "rails/all"` to `config/application.rb`.

This behavior was not tested anywhere in the test suite.

After this PR, `--skip-action-mailer` must also be absent in order for `require "rails/all"` to be in `config/application.rb`.

The code above is testing two things:

- if none of those 4 skip options are present, then `require "rails/all"` will be in `config/application.rb`.
- if the `--skip-action-mailer` is not present, then `config.action_mailer...` will be in the `config/environments` file

Do you think this is confusing? Would you rather have this split into two different testing methods?